File: src/Model/Expression/ArithmeticExpression.java
package Model.Expression;

import DataTypes.MyIDictionary;
import DataTypes.MyIHeap;
import Exception.ExpressionEvaluationException;
import Exception.DivisionByZeroException;
import Exception.InvalidArithmeticOperatorException;

/**
 * Created by vladc on 12.10.2016.
 */
public class ArithmeticExpression extends Expression {
    private Expression expr1;
    private Expression expr2;
    private char       operator;

    public ArithmeticExpression(Expression expr1, Expression expr2, char operator) {
        this.expr1 = expr1;
        this.expr2 = expr2;
        this.operator = operator;
    }

    public int evaluate(MyIDictionary<String, Integer> symTab, MyIHeap heap) throws ExpressionEvaluationException {
        if (operator == '+')
            return (expr1.evaluate(symTab, heap) + expr2.evaluate(symTab, heap));
        else if (operator == '-')
            return (expr1.evaluate(symTab, heap) - expr2.evaluate(symTab, heap));
        else if (operator == '*')
            return (expr1.evaluate(symTab, heap) * expr2.evaluate(symTab, heap));
        else if (operator == '/') {
            if (expr2.evaluate(symTab, heap) == 0)
                throw new DivisionByZeroException();
            return (expr1.evaluate(symTab, heap) / expr2.evaluate(symTab, heap));
        }
        throw new InvalidArithmeticOperatorException();
    }

    @Override
    public String toString() {
        return expr1.toString() + " " + operator + " " + expr2.toString();
    }
}

File: Toy-Language-Interpreter/Model/Expression/ArithmeticExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.DataTypes;
using Toy_Language_Interpreter.MyException;

namespace Toy_Language_Interpreter.Model.Expression
{
    class ArithmeticExpression : AExpression
    {
        private AExpression expr1;
        private AExpression expr2;
        private char op;

        public ArithmeticExpression(AExpression expr1, char op, AExpression expr2)
        {
            this.expr1 = expr1;
            this.expr2 = expr2;
            this.op = op;
        }

        public override int Evaluate(MyIDictionary<string, int> symTab)
        {
            if (op == '+')
                return (expr1.Evaluate(symTab) + expr2.Evaluate(symTab));
            if (op == '-')
                return (expr1.Evaluate(symTab) - expr2.Evaluate(symTab));
            if (op == '*')
                return (expr1.Evaluate(symTab) * expr2.Evaluate(symTab));
            if (op == '/')
            {
                if (expr2.Evaluate(symTab) == 0)
                    throw new DivisionByZeroException();
                return (expr1.Evaluate(symTab) / expr2.Evaluate(symTab));
            }
            throw new InvalidArithmeticOperatorException();
        }

        public override string ToString()
        {
            return expr1.ToString() + " " + op + " " + expr2.ToString();
        }
    }
}
